src/core/server/workspaces/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -150,17 +176,20 @@ export class WorkspaceSavedObjectsClientWrapper {
         options.workspaces = options.workspaces.filter(
           async (workspaceId) =>
             await this.permissionControl.validate(
-              workspaceId,
-              WorkspacePermissionMode.Read,
-              wrapperOptions.request
+              wrapperOptions.request,
+              {
+                type: WORKSPACE_TYPE,
+                id: workspaceId,
+              },
+              this.formatPermissionModeToStringArray(PermissionMode.Read)
             )
         );
       } else {
         options.workspaces = [
           'public',
           ...(await this.permissionControl.getPermittedWorkspaceIds(
-            WorkspacePermissionMode.Read,
-            wrapperOptions.request
+            wrapperOptions.request,
+            this.formatPermissionModeToStringArray(PermissionMode.Read)
```

Comment:
I feel simply pass `[PermissionMode.Read]` is good enough.

Reply:
Agree.